### ISSUE TITLE
netcat probe leads to panic

### ISSUE THREAD

**Issue by pkpowell** [matrix-org/pinecone#30]
**Describe the bug**
`nc address port` leads to panic

**To Reproduce**
1. run a listening instance `./pinecone -listen 0.0.0.0:4567`
2. in a different terminal / from a different machine run `nc ip-address 4567`
3. listener panics
```
panic: io.ReadFull: read tcp [fe80::942c:69ff:fecc:8251%feth1801]:4567->[fe80::94aa:78ff:fe5d:dac4%feth1801]:59994: i/o timeout

goroutine 24 [running]:
main.main.func3()
	/Users/phil/dev/pinecone/cmd/pinecone/main.go:101 +0x258
created by main.main
	/Users/phil/dev/pinecone/cmd/pinecone/main.go:82 +0x265
```
**Expected behavior**
pinecone should ignore the probe

**Desktop (please complete the following information):**
 - Macos

**Additional context**
this is on 351265543ddf6aeb3e4fef41d668eeea9b2d250e

**Comment by jaller94** [comment]
Same with `curl http://address:port`.

Stack trace:
```
panic: io.ReadFull: unexpected EOF

goroutine 115 [running]:
main.main.func3()
	pinecone/cmd/pinecone/main.go:126 +0x256
created by main.main
	pinecone/cmd/pinecone/main.go:107 +0x4bd

```

**Comment by neilalexander** [comment]
This was fixed in 5b805cec494c5cc1b7773980c1a74db2cb07fe35.
